contracts/src/ERC20Destination.sol

```diff
@@ -62,28 +70,36 @@ contract ERC20Destination is IERC20Bridge, TeleporterTokenDestination, ERC20 {
      *
      * @dev See {IERC20Bridge-send}
      */
-    function send(SendTokensInput calldata input, uint256 amount) external nonReentrant {
+    function send(SendTokensInput calldata input, uint256 amount) external {
         _send(input, amount);
     }
 
     /**
-     * @dev See {IERC20-decimals}
+     * @dev See {IERC20Bridge-sendAndCall}
      */
-    function decimals() public view virtual override returns (uint8) {
+    function sendAndCall(SendAndCallInput calldata input, uint256 amount) external {
+        _sendAndCall(input, amount);
+    }
+
+    /**
+     * @dev See {ERC20-decimals}
+     */
+    function decimals() public view override returns (uint8) {
         return _decimals;
     }
 
     /**
      * @dev See {TeleporterTokenDestination-_deposit}
      */
-    function _deposit(uint256 amount) internal virtual override returns (uint256) {
+    function _deposit(uint256 amount) internal override returns (uint256) {
         return SafeERC20TransferFrom.safeTransferFrom(this, amount);
     }
 
     /**
      * @dev See {TeleporterTokenDestination-_withdraw}
      */
-    function _withdraw(address recipient, uint256 amount) internal virtual override {
+    function _withdraw(address recipient, uint256 amount) internal override {
+        emit TokensWithdrawn(recipient, amount);
```

Comment:
think there's double event emissions right now

Reply:
👁️ fixed in [f41d8f2](https://github.com/ava-labs/teleporter-token-bridge/pull/79/commits/f41d8f2ee56dac18ce4f506e82f01650234d9842) in #79